lib/logstash/filters/drop.rb

```diff
@@ -40,5 +40,6 @@ def register
   public
   def filter(event)
     event.cancel if (@percentage == 100 || rand < (@percentage / 100.0))
+    filter_matched(event)
```

Comment:
This logic does not satisfy the requirements stated in the issue that this PR is supposed to solve.
Consider:
```
if (@percentage == 100 || rand < (@percentage / 100.0))
  event.cancel
else
  filter_matched(event)
end
```